airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py

```diff
@@ -49,9 +50,20 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
         except requests.exceptions.RequestException as e:
             return False, e
 
+    def get_reports(self, config):
+        url = f"{PendoPythonStream.url_base}report"
+        auth = SourcePendoPython._get_authenticator(config)
+        try:
+            session = requests.get(url, headers=auth.get_auth_header())
+            body = session.json()
+            return [{"id": obj["id"], "type": obj["type"], "fields": obj["aggregation"]["fields"]} for obj in body]
```

Comment:
Many reports don't have an `aggegation.fields` value-- 45 of the 60 I looked at in the customer's account.  did you look at generating the schemas from the structure of the responses?